src/parquet/column/statistics.h

```diff
@@ -104,14 +104,15 @@ class PARQUET_EXPORT TypedRowGroupStatistics : public RowGroupStatistics {
   std::string EncodedMax() override;
 
  private:
-  MemoryAllocator* allocator_;
-
   bool has_min_max_ = false;
   T min_;
   T max_;
+  MemoryAllocator* allocator_;
 
   void PlainEncode(const T& src, std::string* dst);
   void PlainDecode(const std::string& src, T* dst);
+  void Copy(const T& src, T& dst, std::vector<uint8_t>& free_flag);
+  std::vector<uint8_t> min_buffer_, max_buffer_;
```

Comment:
You can use the `OwnedMutableBuffer` here along with the associated `MemoryAllocator`